webapi/src/test/scala/org/knora/webapi/util/search/gravsearch/GravsearchParserSpec.scala

```diff
@@ -2131,12 +2132,23 @@ object GravsearchParserSpec extends ZIOSpecDefault {
     ),
   )
 
+  // quick fix, because the vector order seems to be non-deterministic
+  def removeConstructClauseStatements(cq: ConstructQuery): ConstructQuery =
+    cq.copy(constructClause = cq.constructClause.copy(statements = Vector()))
+
+  def compareQueriesWithStatementSets(a: ConstructQuery, b: ConstructQuery): TestResult =
+    assertTrue(
+      removeConstructClauseStatements(a) == removeConstructClauseStatements(b),
+      a.constructClause.statements.toSet == b.constructClause.statements.toSet,
```

Comment:
[In SPARQL, order matters.](https://github.com/blazegraph/database/wiki/SPARQL_Order_Matters)

If the order of the statements has changed this would mean the results could be different.

Reply:
Right, but that is in the WHERE clause where the OPTIONAL statements can indeed change the meaning, but these are the statements in the curly braces of a CONSTRUCT, those should be reorderable.

Obviously, the fact that the order changes regardless of input is very weird and unfavorable.

Reply:
Yes, you are right, the order of the statements in the CONSTRUCT clause is not important for the result set, it will still be the same graph.

Reply:
Thank you.